Library/Homebrew/diagnostic.rb

```diff
@@ -835,13 +835,15 @@ def check_deleted_formula
 
         deleted_formulae = kegs.filter_map do |keg|
           tap = Tab.for_keg(keg).tap
+          keg_name = tap ? "#{tap}/#{keg.name}" : keg.name
 
           loadable = [
             Formulary::FromAPILoader,
+            Formulary::FromTapLoader,
             Formulary::FromNameLoader,
           ].any? do |loader_class|
             loader = begin
-              loader_class.try_new(keg.name, warn: false)
+              loader_class.try_new(keg_name, warn: false)
```

Comment:
The check below is now not needed anymore, given that it will load a specific tap now if specified.

Reply:
Looks like we now need to check `.path.exist?` as that was originally handled inside `FromNameLoader`.

Reply:
Actually `FromAPILoader` will have issues if `homebrew-core` is untapped.

Reply:
> Actually `FromAPILoader` will have issues if `homebrew-core` is untapped.

I assume this is now handled by only checking the path for `FromTapLoader`.